### PR TITLE
[WIP] controllers compatible with sf3

### PR DESCRIPTION
**Pull request by amine2z**
This PR fix some (in progress) BC for controller to be compatible with sf3

### CODE REVIEW

File: Controller/AdminResettingController.php [hunk 1]
```diff
@@ -27,7 +27,14 @@ class AdminResettingController extends ResettingController
      */
     public function requestAction()
     {
-        if ($this->container->get('security.context')->isGranted('IS_AUTHENTICATED_FULLY')) {
+        if ($this->container->has('security.authorization_checker')) {
+            $securityService = $this->container->get('security.authorization_checker');
```
Comment (Soullivaneuh) [inline Controller/AdminResettingController.php]: Please simplify this and rename varible to `$authorizationChecker`:

```php
$authorizationChecker = $this->container->has('security.authorization_checker') ? $this->container->get('security.authorization_checker') : $this->container->get('security.context');
```

To be done on other actions.

File: Controller/ChangePasswordFOSUser1Controller.php [hunk 2]
```diff
@@ -34,7 +34,13 @@ class ChangePasswordFOSUser1Controller extends Controller
      */
     public function changePasswordAction()
     {
-        $user = $this->get('security.context')->getToken()->getUser();
+        if ($this->container->has('security.token_storage')) {
+            $user = $this->container->get('security.token_storage')->getToken()->getUser();
```
Comment (Soullivaneuh) [inline Controller/ChangePasswordFOSUser1Controller.php]: Same tricks as https://github.com/sonata-project/SonataUserBundle/pull/662/files#r47201326

**Comment by rande** [comment]
This work duplicate https://github.com/sonata-project/SonataUserBundle/pull/555#issuecomment-134705360 and FosUser 1.X will not support Symfony3.X . I will prefer to have a branch 3.X for SonataUserBundle with the support for FosUser 2.X and Symfony3.X.

**Comment by amine2z** [comment]
@rande can you create the new branch please

**Comment by rande** [comment]
There is a WIP to support FOS2.X on a dedicated branch: https://github.com/sonata-project/SonataUserBundle/tree/add_support_for_fos_user2

**Pull request closed without merging**